src/django_rich/test.py

```diff
@@ -161,18 +184,72 @@ class RichTestRunner(DiscoverRunner.test_runner):  # type: ignore [misc]
     resultclass = RichTextTestResult
 
 
+def positive_int(value: str) -> int:
+    try:
+        num = int(value)
+    except ValueError:
+        raise argparse.ArgumentTypeError("should be an integer")
+    if num < 1:
+        raise argparse.ArgumentTypeError("should be at least 1")
+    return num
+
+
 class RichRunner(DiscoverRunner):
     pdb: bool  # django-stubs missing
 
     test_runner = RichTestRunner
 
+    @classmethod
+    def add_arguments(cls, parser: argparse.ArgumentParser) -> None:
+        super().add_arguments(parser)
+        parser.add_argument(
+            "--slowest",
+            nargs="?",
+            default=20,
+            type=positive_int,
+            metavar="num_slowest",
+            help=(
+                "The number of slow tests and setup/teardown methods to "
+                + "report when --timing is used."
+            ),
+        )
+
+    def __init__(self, *args: Any, slowest: int, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+        self.slowest = slowest
+
     # django-stubs bad return type, fixed in:
     # https://github.com/typeddjango/django-stubs/pull/1069
     def get_resultclass(  # type: ignore [override]
         self,
-    ) -> type[unittest.TextTestResult] | None:
+    ) -> Callable[[], unittest.TextTestResult]:
+        klass: type[RichTextTestResult]
         if self.debug_sql:
-            return RichDebugSQLTextTestResult
+            klass = RichDebugSQLTextTestResult
         elif self.pdb:
-            return RichPDBDebugResult
-        return None
+            klass = RichPDBDebugResult
+        else:
+            klass = RichTextTestResult
+        return partial(klass, slowest=self.slowest)
+
+    def suite_result(  # type: ignore [override]
+        self,
+        suite: unittest.TestSuite,
+        result: RichTextTestResult,
+        **kwargs: Any,
+    ) -> int:
+        time_keeper = self.time_keeper  # type: ignore [attr-defined]
+        if result.collectedDurations and isinstance(time_keeper, TimeKeeper):
+            num = min(len(result.collectedDurations), self.slowest)
+            result.console.print(
+                DJANGO_GREEN_RULE,
+                f"Slowest {num} Test{'s' if num != 1 else ''}",
```

Comment:
TIL. 🤯

Reply:
I wonder  if it's worth mentioning the total number of tests. So something like "Slowest 15 tests (Total tests 20)" 🤔